File: idiom/go/main1.go
package main

import (
	"fmt"
	"io/ioutil"
)

var path = "file.txt"

func printAndGetContent(path string) string {
	content, _ := ioutil.ReadFile(path)
	fmt.Printf("%p", content)
	return string(content)
}

func main() {
	content := printAndGetContent(path)

	fmt.Printf("%s", content)
}

File: idiom/rust/src/main1.rs
use std::fs::read_to_string;

const PATH: &str = "file.txt";

fn print_and_get_content(path: &str) -> String {
    let content_result = read_to_string(path);

    return content_result
    // if let Ok(content) = content_result {
    //     println!("{}", content);
    //     return content
    // } else {
    //     return "EMPTY".to_string()
    // }
}

fn main() {
    let content = print_and_get_content(PATH);

    println!("{}", content);
}
